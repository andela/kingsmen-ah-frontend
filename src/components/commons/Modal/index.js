import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Portal from '@components/commons/utilities/Portal';
import './index.scss';

class Modal extends Component {
  constructor(props) {
    super(props);

    this.state = {

    };
  }

  render() {
    const {
      children, title, toggle, exitModal
    } = this.props;
    return (
      <Portal>
        <div className={classNames('modal flex', { 'is-visible': toggle })}>
          <div className="modal-container text-center relative justify-center items-center">
            <div role="presentation" className="absolute modal-exit bg-white rounded-full py-1 px-3 shadow-lg cursor-pointer flex hover:bg-gray-300" onClick={exitModal}>
              <span className="icon ion-md-cancel">x</span>
            </div>
            <div className="modal_title">
              <h2 className="text-normal font-bold text-xl">{title}</h2>
            </div>
            <div className="modal_body">
              {children}
            </div>
          </div>
        </div>
      </Portal>
    );
  }
}

Modal.propTypes = {
  toggle: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  exitModal: PropTypes.func.isRequired
};

Modal.defaultProp = {
  toggle: false,
  children: ''
};


export default Modal;
