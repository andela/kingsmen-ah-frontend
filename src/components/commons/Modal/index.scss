$modal-bg: rgba(52, 54, 66, 0.9);

.modal {
  background: $modal-bg;
  cursor: pointer;
  position: fixed;
  height: 100%;
  top: 0;
  left: 0;
  overflow-y: auto;
  opacity: 0;
  transition: opacity 0.3s 0, visibility 0 0.3s;
  visibility: hidden;
  width: 100%;
  z-index: 21;

  &.is-visible {
    opacity: 1;
    transition: opacity 0.3s 0, visibility 0 0;
    visibility: visible;
  }

  &.modal-container {
    transform: translateY(0);
  }
}

.modal-container {
  background: #fff;
  border-radius: 0.25em;
  cursor: auto;
  margin: 3em auto 4em;
  position: relative;
  padding: 30px;
  width: 70%;
  z-index: 1000;
}

.modal-exit {
  right: -10px;
  top: -10px;
}
