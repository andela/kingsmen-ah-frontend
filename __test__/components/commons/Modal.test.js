import React from 'react';
import Modal from '@components/commons/Modal';


const props = {
    title: 'New Modal',
    toggle: false,
    exitModal: jest.fn()
};

const mockClick = jest.fn(props.toggle = false);

describe('<Modal />', () => {
    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'app');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);


    it('should render a modal correctly', () => {
        const shallowWrapper = mount(<Modal {...props}><h1>Hello World</h1></Modal>);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Modal')).toBeTruthy();
        shallowWrapper.unmount();
    });

    it('should exit the toggle when exit is clicked', () => {
      const shallowWrapper = mount(<Modal {...props} exitModal={mockClick}><h1>Hello World</h1></Modal>);

      expect(shallowWrapper.props().toggle).toEqual(false);
    });
});
