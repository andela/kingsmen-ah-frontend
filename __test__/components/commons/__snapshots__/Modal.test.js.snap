// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render a modal correctly 1`] = `
<Modal
  exitModal={[MockFunction]}
  title="New Modal"
  toggle={false}
>
  <Portal>
    <Portal
      containerInfo={
        <div
          id="app"
        >
          <div
            class="modal flex"
          >
            <div
              class="modal-container text-center relative justify-center items-center"
            >
              <div
                class="absolute modal-exit bg-white rounded-full py-1 px-3 shadow-lg cursor-pointer flex hover:bg-gray-300"
                role="presentation"
              >
                <span
                  class="icon ion-md-cancel"
                >
                  x
                </span>
              </div>
              <div
                class="modal_title"
              >
                <h2
                  class="text-normal font-bold text-xl"
                >
                  New Modal
                </h2>
              </div>
              <div
                class="modal_body"
              >
                <h1>
                  Hello World
                </h1>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <div
        className="modal flex"
      >
        <div
          className="modal-container text-center relative justify-center items-center"
        >
          <div
            className="absolute modal-exit bg-white rounded-full py-1 px-3 shadow-lg cursor-pointer flex hover:bg-gray-300"
            onClick={[MockFunction]}
            role="presentation"
          >
            <span
              className="icon ion-md-cancel"
            >
              x
            </span>
          </div>
          <div
            className="modal_title"
          >
            <h2
              className="text-normal font-bold text-xl"
            >
              New Modal
            </h2>
          </div>
          <div
            className="modal_body"
          >
            <h1>
              Hello World
            </h1>
          </div>
        </div>
      </div>
    </Portal>
  </Portal>
</Modal>
`;
